crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -1,87 +1,135 @@
 use crate::config::EthereumSettlementConfig;
 use crate::conversion::{slice_slice_u8_to_vec_u256, slice_u8_to_u256};
-use crate::types::EthHttpProvider;
+pub mod clients;
+pub mod config;
+pub mod conversion;
+
+#[cfg(test)]
+lazy_static! {
+    static ref SHOULD_IMPERSONATE_ACCOUNT: bool = get_env_var_or_panic("TEST_IMPERSONATE_OPERATOR") == *"1";
+    static ref TEST_DUMMY_CONTRACT_ADDRESS: String = get_env_var_or_panic("TEST_DUMMY_CONTRACT_ADDRESS");
+    static ref ADDRESS_TO_IMPERSONATE: Address =
+        Address::from_str("0x2C169DFe5fBbA12957Bdd0Ba47d9CEDbFE260CA7").expect("Unable to parse address");
+    static ref TEST_NONCE: u64 = 666068;
+}
+
+#[cfg(test)]
+mod tests;
+pub mod types;
+
+#[cfg(test)]
+use {alloy::providers::RootProvider, alloy::transports::http::Http, reqwest::Client};
+
+#[cfg(not(test))]
+use types::EthHttpProvider;
 
 pub const ENV_PRIVATE_KEY: &str = "ETHEREUM_PRIVATE_KEY";
 
 lazy_static! {
     pub static ref CURRENT_PATH: PathBuf = std::env::current_dir().unwrap();
-    pub static ref KZG_SETTINGS: KzgSettings = KzgSettings::load_trusted_setup_file(
-        CURRENT_PATH.join("../../../orchestrator/src/jobs/state_update_job/trusted_setup.txt").as_path()
-    )
-    .expect("Error loading trusted setup file");
+    pub static ref KZG_SETTINGS: KzgSettings =
+        // TODO: set more generalized path
+        KzgSettings::load_trusted_setup_file(CURRENT_PATH.join("src/trusted_setup.txt").as_path())
+            .expect("Error loading trusted setup file");
 }
 
 #[allow(dead_code)]
 pub struct EthereumSettlementClient {
-    provider: Arc<EthHttpProvider>,
     core_contract_client: StarknetValidityContractClient,
     wallet: EthereumWallet,
     wallet_address: Address,
+    #[cfg(not(test))]
+    provider: Arc<EthHttpProvider>,
+    #[cfg(test)]
+    provider: RootProvider<Http<Client>>,
 }
 
 impl EthereumSettlementClient {
+    #[cfg(not(test))]
     pub fn with_settings(settings: &impl SettingsProvider) -> Self {
         let settlement_cfg: EthereumSettlementConfig = settings.get_settings(SETTLEMENT_SETTINGS_NAME).unwrap();
 
         let private_key = get_env_var_or_panic(ENV_PRIVATE_KEY);
         let signer: PrivateKeySigner = private_key.parse().expect("Failed to parse private key");
-        let wallet = EthereumWallet::from(signer.clone());
-
         let wallet_address = signer.address();
+        let wallet = EthereumWallet::from(signer);
 
         let provider = Arc::new(
             ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(settlement_cfg.rpc_url),
         );
         let core_contract_client = StarknetValidityContractClient::new(
-            Address::from_str(&settlement_cfg.core_contract_address)
-                .expect("Failed to convert the validity contract address.")
-                .0
-                .into(),
+            Address::from_str(&settlement_cfg.core_contract_address).unwrap().0.into(),
```

Comment:
why remove `.expect`?

Reply:
You are correct, this was removed by mistake, added.